### PR TITLE
[extension/opamp] Add config options to specify extra non-identifying attributes

### LINKED ISSUE #32107: [extension/opamp] Add support for user-specified non-identifying attributes
### Component(s)

extension/opamp

### Is your feature request related to a problem? Please describe.

In the OpAMP spec, [non-identifying attributes](https://github.com/open-telemetry/opamp-spec/blob/main/specification.md#agentdescriptionnon_identifying_attributes) include "any user-defined attributes that the end user would like to associate with this Agent."

We would like to be able to add some extra attributes to the agent description this way.

### Describe the solution you'd like

I think a simple solution would be to just have a section of the config that unmarshals to map[string]string, where the attributes can be specified:

```yaml
extension:
  opamp:
    non_identifying_attributes:
      custom.attribute: value
      another.custom.attribute: "another value"
```

If one of the automatically filled non identifying attributes are specified (e.g. `host.name`, `host.arch`, `os.type`), I think the user-specified value of the attribute should act as an override for these attributes.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by BinaryFissionGames**
**Description:** <Describe what has changed.>
* Adds a new `agent_description.non_identifying_attributes` config option to allow setting user-defined non-identifying attributes

**Link to tracking Issue:** Closes #32107

**Testing:**
Added unit tests
Manually tested against an OpAMP server

**Documentation:**
Added new parameter to extension docs

### CODE REVIEW

File: extension/opampextension/README.md [hunk 1]
```diff
@@ -30,6 +30,8 @@ The following settings are optional:
   instance UID remains constant across process restarts.
 - `capabilities`: Keys with boolean true/false values that enable a particular OpAMP capability.
   - `reports_effective_config`: Whether to enable the OpAMP ReportsEffectiveConfig capability. Default is `true`.
+- `agent_description`: Modify the agent description reported to the OpAMP server.
+  - `non_identifying_attributes`: A map of key value pairs that will be added to the NonIdentifyingAttributes reported to the OpAMP server.
```
[the remark this answers is not in the document]
Comment (BinaryFissionGames) [reply extension/opampextension/README.md]: Good point, that should just be written in plain english. I've also linked to the relevant section of the spec, and added a note about the behavior with the default values.

File: extension/opampextension/README.md [hunk 2]
```diff
@@ -30,6 +30,8 @@ The following settings are optional:
   instance UID remains constant across process restarts.
 - `capabilities`: Keys with boolean true/false values that enable a particular OpAMP capability.
   - `reports_effective_config`: Whether to enable the OpAMP ReportsEffectiveConfig capability. Default is `true`.
+- `agent_description`: Setting that modify the agent description reported to the OpAMP server.
```
Comment (crobert-1) [inline extension/opampextension/README.md:33]:
```suggestion
- `agent_description`: Setting that modifies the agent description reported to the OpAMP server.
```
Nit

File: extension/opampextension/README.md [hunk 3]
```diff
@@ -30,6 +30,8 @@ The following settings are optional:
...
+  - `non_identifying_attributes`: A map of key value pairs that will be added to the [non-identifying attributes](https://github.com/open-telemetry/opamp-spec/blob/main/specification.md#agentdescriptionnon_identifying_attributes) reported to the OpAMP server. If an attribute collides with the default non identifying attributes that are automatically added, the ones specified here take precedence.
```
Comment (crobert-1) [inline extension/opampextension/README.md:34]:
```suggestion
  - `non_identifying_attributes`: A map of key value pairs that will be added to the [non-identifying attributes](https://github.com/open-telemetry/opamp-spec/blob/main/specification.md#agentdescriptionnon_identifying_attributes) reported to the OpAMP server. If an attribute collides with the default non-identifying attributes that are automatically added, the ones specified here take precedence.
```
Nit for consistency in how `non-identifying attribute` is written.

**Review by crobert-1 — APPROVED** [review]
Just a couple wording nits, functionality looks good to me 👍

**Review by JaredTan95 — APPROVED** [review]
I happened to have this requirement, thank you for your tribute
LGTM

**Review by djaglowski — APPROVED** [review]
